chatsky/context_storages/database.py

```diff
@@ -8,181 +8,156 @@
 This class implements the basic functionality and can be extended to add additional features as needed.
 """
 
-import asyncio
-import importlib
-import threading
-from functools import wraps
 from abc import ABC, abstractmethod
-from typing import Callable, Hashable, Optional
+from asyncio import Lock
+from functools import wraps
+from importlib import import_module
+from pathlib import Path
+from typing import Any, Awaitable, Callable, Dict, List, Literal, Optional, Set, Tuple, Union
 
 from .protocol import PROTOCOLS
-from chatsky.core import Context
-
 
-class DBContextStorage(ABC):
-    r"""
-    An abstract interface for `chatsky` DB context storages.
-    It includes the most essential methods of the python `dict` class.
-    Can not be instantiated.
-
-    :param path: Parameter `path` should be set with the URI of the database.
-        It includes a prefix and the required connection credentials.
-        Example: postgresql+asyncpg://user:password@host:port/database
-        In the case of classes that save data to hard drive instead of external databases
-        you need to specify the location of the file, like you do in sqlite.
-        Keep in mind that in Windows you will have to use double backslashes '\\'
-        instead of forward slashes '/' when defining the file path.
+_SUBSCRIPT_TYPE = Union[Literal["__all__"], int, Set[str]]
+_SUBSCRIPT_DICT = Dict[str, Union[_SUBSCRIPT_TYPE, Literal["__none__"]]]
 
-    """
 
-    def __init__(self, path: str):
+class DBContextStorage(ABC):
+    _main_table_name: Literal["main"] = "main"
+    _turns_table_name: Literal["turns"] = "turns"
+    _key_column_name: Literal["key"] = "key"
+    _id_column_name: Literal["id"] = "id"
+    _current_turn_id_column_name: Literal["current_turn_id"] = "current_turn_id"
+    _created_at_column_name: Literal["created_at"] = "created_at"
+    _updated_at_column_name: Literal["updated_at"] = "updated_at"
+    _misc_column_name: Literal["misc"] = "misc"
+    _framework_data_column_name: Literal["framework_data"] = "framework_data"
+    _labels_field_name: Literal["labels"] = "labels"
+    _requests_field_name: Literal["requests"] = "requests"
+    _responses_field_name: Literal["responses"] = "responses"
+    _default_subscript_value: int = 3
+
+    def __init__(
```

Comment:
An idea about removing `asyncio.run` from init:
add `connect` method to `DBContextStorage` that will be called in `pipeline.run` and also in every other db method.

Reply:
I meant in the `pipeline.run` method (which currently calls `messenger_interface.connect` and is an entry point for every chatsky bot), not `pipeline._run_pipeline`.

Which is what I meant when I said that it isn't that lazy:
DB will be initialized before the first request is received.